astropy/coordinates/angles/formats.py

```diff
@@ -399,42 +399,28 @@ def parse_angle(angle, unit=None, debug=False):
     return _AngleParser().parse(angle, unit, debug=debug)
 
 
-def degrees_to_dms(d):
+def decimal_to_sexagesimal(a, /):
     """
-    Convert a floating-point degree value into a ``(degree, arcminute,
-    arcsecond)`` tuple.
+    Convert a floating-point input to a 3 tuple
+    - if input is in degrees, the result is (degree, arcminute, arcsecond)
+    - if input is in hourangle, the result is (hour, minute, second)
     """
-    sign = np.copysign(1.0, d)
+    sign = np.copysign(1.0, a)
+    # assuming a in degree, these are (degree fraction, degree)
+    (df, d) = np.modf(np.fabs(a))
 
-    (df, d) = np.modf(np.abs(d))  # (degree fraction, degree)
-    (mf, m) = np.modf(df * 60.0)  # (minute fraction, minute)
+    # assuming a in degree, these are (arcminute fraction, arcminute)
+    (mf, m) = np.modf(df * 60.0)
     s = mf * 60.0
 
     return np.floor(sign * d), sign * np.floor(m), sign * s
 
 
-def hours_to_hms(h):
+def decimal_to_string(angle, precision=None, pad=False, sep=(":",), fields=3):
```

Comment:
```suggestion
def _decimal_to_sexagesimal_string(angle, precision=None, pad=False, sep=(":",), fields=3):
```

Reply:
done !

Reply:
I didn't point this out in any text, but the name of this function should still include "sexagesimal" somewhere.

Reply:
sorry I missed it ! done